src/System.Management.Automation/engine/SecurityDescriptorCommandsHelperBase.cs

```diff
@@ -0,0 +1,502 @@
+        /// </summary>
+        /// <param name="instance">
+        /// The PSObject for which to obtain the central access policy ID.
+        /// </param>
+        /// <returns>
+        /// The central access policy ID of the provided PSObject.
+        /// </returns>
+        public static SecurityIdentifier? GetCentralAccessPolicyId(PSObject instance)
+        {
+            SessionState sessionState = new();
+            string path = sessionState.Path.GetUnresolvedProviderPathFromPSPath(
+                GetPath(instance));
+            IntPtr pSd = IntPtr.Zero;
+
+            try
+            {
+                // Get the file's SACL containing the CAPID ACE.
+                uint rs = NativeMethods.GetNamedSecurityInfo(
+                    path,
+                    NativeMethods.SeObjectType.SE_FILE_OBJECT,
+                    NativeMethods.SecurityInformation.SCOPE_SECURITY_INFORMATION,
+                    out IntPtr pOwner,
+                    out IntPtr pGroup,
+                    out IntPtr pDacl,
+                    out IntPtr pSacl,
+                    out pSd);
+                if (rs != NativeMethods.ERROR_SUCCESS)
+                {
+                    throw new Win32Exception((int)rs);
+                }
+
+                if (pSacl == IntPtr.Zero)
+                {
+                    return null;
+                }
+
+                NativeMethods.ACL sacl = Marshal.PtrToStructure<NativeMethods.ACL>(pSacl);
+                if (sacl.AceCount == 0)
+                {
+                    return null;
+                }
+
+                // Extract the first CAPID from the SACL that does not have INHERIT_ONLY_ACE flag set.
+                IntPtr pAce = pSacl + Marshal.SizeOf(new NativeMethods.ACL());
+                for (ushort aceIdx = 0; aceIdx < sacl.AceCount; aceIdx++)
+                {
+                    NativeMethods.ACE_HEADER ace = Marshal.PtrToStructure<NativeMethods.ACE_HEADER>(pAce);
+                    Dbg.Diagnostics.Assert(ace.AceType ==
+                        NativeMethods.SYSTEM_SCOPED_POLICY_ID_ACE_TYPE,
+                        "Unexpected ACE type: " + ace.AceType.ToString(CultureInfo.CurrentCulture));
+                    if ((ace.AceFlags & NativeMethods.INHERIT_ONLY_ACE) == 0)
+                    {
+                        break;
+                    }
+
+                    pAce += ace.AceSize;
+                }
+
+                IntPtr pSid = pAce + Marshal.SizeOf(new NativeMethods.SYSTEM_AUDIT_ACE()) -
+                    Marshal.SizeOf(new uint());
+                bool ret = NativeMethods.IsValidSid(pSid);
+                if (!ret)
+                {
+                    throw new Win32Exception(Marshal.GetLastWin32Error());
+                }
+
+                return new SecurityIdentifier(pSid);
+            }
+            finally
+            {
+                NativeMethods.LocalFree(pSd);
+            }
+        }
+
+#if !CORECLR
```

Comment:
Makes sense to remove the code?

Reply:
Or enable for IsWindowsDesktop?

Reply:
The code shouldn't be removed. It was enclosed by `CORECLR` because the Win32 API `LsaQueryCAPs` was not available on NanoServer. Now it should be enabled in theory. I think we need @TravisEz13 or @PaulHigin to take a look.

Reply:
Thanks! Docs say the API is on Windows 8/2012+ desktop apps only.
https://docs.microsoft.com/en-us/windows/win32/api/ntlsa/nf-ntlsa-lsaquerycaps